python/cudf/cudf/core/column/column.py

```diff
@@ -119,25 +115,19 @@ def __repr__(self):
     def to_pandas(
         self, index: ColumnLike = None, nullable: bool = False, **kwargs
     ) -> "pd.Series":
-        if nullable and self.dtype in cudf_dtypes_to_pandas_dtypes:
-            pandas_nullable_dtype = cudf_dtypes_to_pandas_dtypes[self.dtype]
-            arrow_array = self.to_arrow()
-            pandas_array = pandas_nullable_dtype.__from_arrow__(arrow_array)
-            pd_series = pd.Series(pandas_array, copy=False)
-        elif str(self.dtype) in NUMERIC_TYPES and self.null_count == 0:
-            pd_series = pd.Series(cupy.asnumpy(self.values), copy=False)
-        elif is_interval_dtype(self.dtype):
-            pd_series = pd.Series(
-                pd.IntervalDtype().__from_arrow__(self.to_arrow())
-            )
-        else:
-            pd_series = self.to_arrow().to_pandas(**kwargs)
+        """Convert object to pandas type.
+
+        The default implementation falls back to PyArrow for the conversion.
+        """
+        pd_series = self.to_arrow().to_pandas(**kwargs)
 
         if index is not None:
             pd_series.index = index
         return pd_series
 
     def __iter__(self):
+        # TODO: Why don't we just implement this method in terms of one of the
+        # proposed alternatives (to_arrow, to_pandas, or values_host)?
```

Comment:
We specifically don't want to allow implicit iteration over GPU-backed cuDF objects (for performance reasons).

Reply:
I'll remove this if there are any review comments on this PR, otherwise I'll remove in the next PR in the chain once this one is merged.

Reply:
Sounds good, will remove.